### DISCUSSION TITLE
Valkey for Production as a replacement for Redis

### DISCUSSION THREAD

**Discussion by Susmit07** [valkey-io/valkey#1270] [Q&A]
Hello Community Members,

We were using Redis till now in our production servers, but it's no more open-sourced as we all know

So thought of exploring Redis.

Our existing setup is something like this

<img width="798" alt="image" src="https://github.com/user-attachments/assets/942a1193-597e-4a2e-9bb3-82f0b8ef1fea">

We have 1 master 5 slave and 3 sentinel nodes (Kubernetes deployment). Authentication we used Password based within cluster-IP

Does Valkey has the concept of sentinel pods to monitor the master nodes during failover. 

What will be the ideal way of deployment for our use-case, with Master-Slave mode of deployment.

Will be awaiting your kind reply and guidance. Is there any Kubernetes operator we can use to fast pace the mode of deployment

Durability is a top priority for us, if the kubernetes pod crashes Valkey must ensure data is recoverable after a restart, as we are building Metadata service backed by a key value store, so if somehow data gets lost the entire system goes for a toss

Thanks,
Susmit

**Comment by amirreza-sf80** [comment]
hi
valkey is a fork of redis
in most senses (like 98%) it's the same thing
has the same tools (again, mostly)
works the same way

you can make sentinels, clusters, single servers, use passwords and so on

**Reply by aq-hue** [reply]
@amirreza8002 

> valkey is a fork of redis
> in most senses (like 98%) it's the same thing
> has the same tools (again, mostly)
> works the same way
> 
> you can make sentinels, clusters, single servers, use passwords and so on

Even with Valkey engine version 8?

I have a docker container which is running version 8 and now I can't connect to it using Datagrip with password enable.
Valkey engine version 7.2 work normally as expected, so if you know please tell me how can I use Datagrip or any GUI client
to connect to Valkey engine version 8. I know, I still can connect to the Valkey by using CLI
but GUI client provide way better DX when dev and debugging. And for us, this problem is only thing which make
my company can't completely switch to Valkey engine version 8.

**Reply by amirreza-sf80** [reply]
hi 
of the top of my head i don't remember what changed about auth on version 8, and i haven't used the tool you mentioned 

I'd say make a new discussion, and provide more information, such as the errors you get and so on

**Reply by aq-hue** [reply]
@amirreza8002 

Hi. Got it, I will make a new discussion using my personal Github account.

**Comment by martinrvisser** [comment]
Have you checked protected-mode ?
